interfaces/builtin/mir.go

```diff
@@ -91,6 +93,22 @@ unix (receive, send) type=seqpacket addr=none peer=(label=###SLOT_SECURITY_TAGS#
 deny /{dev,run,var/run}/shm/lttng-ust-* rw,
 `
 
+const mirPermanentSlotUdev = `
+# This file contains udev rules for mir server.
+#
+# Do not edit this file, it will be overwritten on updates
+
+KERNEL=="tty",       TAG+="%[1]s"
+KERNEL=="tty[0-9]*", TAG+="%[1]s"
+
+# input devices
+KERNEL=="mice",   NAME="input/%%k", TAG+="%[1]s"
+KERNEL=="mouse*", NAME="input/%%k", TAG+="%[1]s"
+KERNEL=="event*", NAME="input/%%k", TAG+="%[1]s"
+KERNEL=="js*",    NAME="input/%%k", TAG+="%[1]s"
+KERNEL=="ts*",    NAME="input/%%k", TAG+="%[1]s"
```

Comment:
RE: /dev/input/ts*
1.https://kernel.googlesource.com/pub/scm/linux/hotplug/udev/+/127/rules/rules.d/50-udev-default.rules#27
2.https://github.com/kergoth/tslib
It's a udev rule for the touchscreen device.